### PR TITLE
StaticLLMPipeline: Enable conditional L0 allocation

### PR DESCRIPTION
**Pull request by TolyaTalamanov**

### CODE REVIEW

File: src/cpp/src/llm_pipeline_static.cpp [hunk 1]
```diff
@@ -227,7 +227,12 @@ ov::AnyMap get_baseline_common_config() {
 
 ov::AnyMap get_default_common_config(const std::shared_ptr<ov::Model>& model) {
     auto config = get_baseline_common_config();
-    config.emplace("NPUW_WEIGHTS_BANK_ALLOC", "CPU");
+    auto npu_l0 = std::getenv("OPENVINO_GENAI_NPU_L0");
+    if (npu_l0 != nullptr && std::stoi(npu_l0) == 0) {
+        config.emplace("NPUW_FUNCALL_FOR_ALL", "YES");
```
Comment (TolyaTalamanov) [inline src/cpp/src/llm_pipeline_static.cpp:232]: Must be vice-versa

Comment (TolyaTalamanov) [reply src/cpp/src/llm_pipeline_static.cpp]: Done

File: src/cpp/src/llm_pipeline_static.cpp [hunk 2]
```diff
@@ -227,7 +227,12 @@ ov::AnyMap get_baseline_common_config() {
 
 ov::AnyMap get_default_common_config(const std::shared_ptr<ov::Model>& model) {
     auto config = get_baseline_common_config();
-    config.emplace("NPUW_WEIGHTS_BANK_ALLOC", "CPU");
+    const char* npu_l0 = std::getenv("DISABLE_OPENVINO_GENAI_NPU_L0");
+    if (npu_l0 && std::atoi(npu_l0) == 1) {
+        config.emplace("NPUW_WEIGHTS_BANK_ALLOC", "CPU");
+    } else {
+        config.emplace("NPUW_FUNCALL_FOR_ALL", "YES");
```
Comment (TolyaTalamanov) [inline src/cpp/src/llm_pipeline_static.cpp:234]: This path causing issues on CPU - inverstigating

File: src/cpp/src/llm_pipeline_static.cpp [hunk 3]
```diff
@@ -339,8 +360,8 @@ void StaticLLMPipeline::setupAndCompileModels(
     m_prefill_model = m_kvcache_model->clone();
     m_prefill_model->set_friendly_name(m_kvcache_model->get_friendly_name() + "_prefill");
     // (7) Reshape both models to static shape
-    const auto kMaxPromptLen = pop_or_default(pipeline_config, "MAX_PROMPT_LEN", 1024u);
-    const auto kMinResponseLen = pop_or_default(pipeline_config, "MIN_RESPONSE_LEN", 150u);
+    const uint32_t kMaxPromptLen = pop_int_and_cast(pipeline_config, "MAX_PROMPT_LEN").value_or(1024u);
```
Comment (TolyaTalamanov) [inline src/cpp/src/llm_pipeline_static.cpp:363]: Small fix for extracting `MAX_PROMPT_LEN` and `MIN_RESPONSE_LEN` properties

**Review by dmatveev — APPROVED** [review]

**Pull request merged**